src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -186,19 +231,27 @@ protected T WithOptions(ApplicationOptions applicationOptions)
         }
 
         /// <summary>
+        /// Sets the identifier of the software component (libraries/SDK) consuming MSAL.NET.
+        /// This will allow for disambiguation between MSAL usage by the app vs MSAL usage 
+        /// by component libraries. You can, for instance set it to the name of your application.
+        /// This is used in telemetry.
         /// </summary>
-        /// <param name="component"></param>
-        /// <returns></returns>
+        /// <param name="component">identifier of the software component (libraries/SDK) consuming MSAL.NET</param>
+        /// <returns>The builder to chain the .With methods</returns>
         public T WithComponent(string component)
```

Comment:
Could we possibly change this to "WithComponentName" or something?  Component is so generic by itself...

Reply:
Can we use something besides "component"?

---
In reply to: [248388802](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/790#discussion_r248388802) [](ancestors = 248388802)

Reply:
WithCallingComponentName ? @shoatman, what do you think?